ansible/playbooks/roles/repository/files/download-requirements/src/mode/red_hat_family_mode.py

```diff
@@ -66,9 +66,8 @@ def _install_base_packages(self):
         self.__remove_dnf_cache_for_custom_repos()
         self._tools.dnf.makecache(True)
 
-        # tar does not come by default from image. We install it, but don't want to remove it
-        if not self._tools.rpm.is_package_installed('tar'):
-            self._tools.dnf.install('tar')
```

Comment:
```
[root@178ae7e372f8 /]# /usr/libexec/platform-python /shared/build/prepare_scripts_rhel_8_x86_64/download-requirements.py /shared/build/prepare_scripts_rhel_8_x86_64/download --log-level debug rhel-8
2022-06-02 08:21:39,031 [DEBUG]: Found Matching OS: `RHEL`
2022-06-02 08:21:39,031 [INFO]: Info summary:
--------------------------------------------------
OS Arch: x86_64
OS Type: rhel-8
Script location: /shared/build/prepare_scripts_rhel_8_x86_64
Directories used:
- files:              /shared/build/prepare_scripts_rhel_8_x86_64/download/files
- grafana dashboards: /shared/build/prepare_scripts_rhel_8_x86_64/download/grafana_dashboards
- images:             /shared/build/prepare_scripts_rhel_8_x86_64/download/images
- packages:           /shared/build/prepare_scripts_rhel_8_x86_64/download/packages
Repos backup file: /var/tmp/enabled-system-repos.tar
Log file location: /download-requirements.log
Retries count: 3
--------------------------------------------------
2022-06-02 08:21:39,156 [INFO]: PyYAML not found, trying to install python3-pyyaml package...
2022-06-02 08:21:39,157 [DEBUG]: [1/3] Running: `dnf install -y python3-pyyaml`
2022-06-02 08:22:16,338 [INFO]: Done.
2022-06-02 08:22:16,338 [DEBUG]: Installed python3-pyyaml package
2022-06-02 08:22:16,338 [DEBUG]: Rerunning as: `/usr/libexec/platform-python /shared/build/prepare_scripts_rhel_8_x86_64/download-requirements.py /shared/build/prepare_scripts_rhel_8_x86_64/download --log-level debug rhel-8 --rerun --pyyaml-installed`
2022-06-02 08:22:16,669 [DEBUG]: Found Matching OS: `RHEL`
2022-06-02 08:22:16,993 [INFO]: Installing base packages...
2022-06-02 08:22:16,993 [DEBUG]: [1/3] Running: `dnf config-manager --dump-variables`
2022-06-02 08:22:19,435 [DEBUG]: [1/3] Running: `dnf update -y libmodulemd`
2022-06-02 08:22:27,768 [DEBUG]: [1/3] Running: `rpm --query --quiet epel-release`
2022-06-02 08:22:27,795 [DEBUG]: [1/3] Running: `dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm`
2022-06-02 08:22:35,393 [DEBUG]: [1/3] Running: `dnf makecache timer -y`
2022-06-02 08:22:47,739 [DEBUG]: [1/3] Running: `dnf install -y ca-certificates`
2022-06-02 08:22:53,366 [DEBUG]: [1/3] Running: `rpm --query --quiet curl`
2022-06-02 08:22:53,373 [DEBUG]: [1/3] Running: `rpm --query --quiet python3-dnf-plugins-core`
2022-06-02 08:22:53,378 [DEBUG]: [1/3] Running: `rpm --query --quiet wget`
2022-06-02 08:22:53,399 [DEBUG]: [1/3] Running: `dnf install -y wget`
2022-06-02 08:23:03,583 [INFO]: Done installing base packages.
2022-06-02 08:23:03,584 [DEBUG]: Creating backup for system repositories...
2022-06-02 08:23:03,584 [DEBUG]: [1/1] Running: `tar -cvf /var/tmp/enabled-system-repos.tar --directory / --verify /etc/yum.repos.d`
Traceback (most recent call last):
  File "/shared/build/prepare_scripts_rhel_8_x86_64/download-requirements.py", line 89, in <module>
    sys.exit(main(sys.argv))
  File "/shared/build/prepare_scripts_rhel_8_x86_64/download-requirements.py", line 72, in main
    run(config)
  File "/shared/build/prepare_scripts_rhel_8_x86_64/src/run.py", line 19, in run
    MODES[config.os_type](config).run()
  File "/shared/build/prepare_scripts_rhel_8_x86_64/src/mode/base_mode.py", line 251, in run
    self._create_backup_repositories()
  File "/shared/build/prepare_scripts_rhel_8_x86_64/src/mode/red_hat_family_mode.py", line 47, in _create_backup_repositories
    verify=True)
  File "/shared/build/prepare_scripts_rhel_8_x86_64/src/command/tar.py", line 69, in pack
    self.run([''.join(short_flags)] + tar_params)
  File "/shared/build/prepare_scripts_rhel_8_x86_64/src/command/command.py", line 52, in run
    process = subprocess.run(process_args, **additional_args)
  File "/usr/lib64/python3.6/subprocess.py", line 423, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib64/python3.6/subprocess.py", line 729, in __init__
    restore_signals, start_new_session)
  File "/usr/lib64/python3.6/subprocess.py", line 1364, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'tar': 'tar'
```

Reply:
Great catch! The motivation for this change is that we should try to avoid changing state (especially implicitly) on the downloading host (it can be any machine). That's why I moved installing tar to Ansible role but I missed to check the base packages list for RHEL :( looking only for Debian where tar was present. Added tar also for RHEL in a new commit.